data-model/src/main/scala/za/co/absa/enceladus/model/MappingTable.scala

```diff
@@ -15,39 +15,78 @@
 
 package za.co.absa.enceladus.model
 
-import java.time.ZonedDateTime
-import za.co.absa.enceladus.model.dataFrameFilter.DataFrameFilter
+import com.fasterxml.jackson.annotation.JsonIgnore
 import com.fasterxml.jackson.databind.node.ArrayNode
-import za.co.absa.enceladus.model.versionedModel.VersionedModel
-import za.co.absa.enceladus.model.backend.audit._
+import io.swagger.v3.oas.annotations.media.{ArraySchema, Schema => AosSchema}
 import za.co.absa.enceladus.model.backend.Reference
+import za.co.absa.enceladus.model.backend.audit._
+import za.co.absa.enceladus.model.dataFrameFilter.DataFrameFilter
+import za.co.absa.enceladus.model.versionedModel.VersionedModel
+
+import java.time.ZonedDateTime
+import scala.annotation.meta.field
+import scala.beans.BeanProperty
+
+case class MappingTable(
+  @(AosSchema@field)(example = "datasetA")
+  @BeanProperty name: String,
+
+  @(AosSchema@field)(example = "1")
+  @BeanProperty version: Int = 1,
+
+  @(AosSchema@field)(implementation = classOf[String], example = "MT description")
+  @BeanProperty description: Option[String] = None,
+
+  @(AosSchema@field)(example = "/input/path/for/mappingtable")
+  @BeanProperty hdfsPath: String,
+
+  @(AosSchema@field)(example = "schemaA")
+  @BeanProperty schemaName: String,
+
+  @(AosSchema@field)(example = "1")
+  @BeanProperty schemaVersion: Int,
+
+  @(ArraySchema@field)(schema = new AosSchema(implementation = classOf[DefaultValue]))
+  @BeanProperty defaultMappingValues: List[DefaultValue] = List(),
+
+  @BeanProperty dateCreated: ZonedDateTime = ZonedDateTime.now(),
+  @(AosSchema@field)(example = "user1")
+  @BeanProperty userCreated: String = null, //scalastyle:ignore null
+
+  @BeanProperty lastUpdated: ZonedDateTime = ZonedDateTime.now(),
+  @(AosSchema@field)(example = "user2")
+  @BeanProperty userUpdated: String = null, //scalastyle:ignore null
 
-case class MappingTable(name: String,
-                        version: Int = 1,
-                        description: Option[String] = None,
+  @(AosSchema@field)(example = "false")
+  @BeanProperty disabled: Boolean = false,
 
-                        hdfsPath: String,
+  @(AosSchema@field)(implementation = classOf[ZonedDateTime])
+  @BeanProperty dateDisabled: Option[ZonedDateTime] = None,
 
-                        schemaName: String,
-                        schemaVersion: Int,
+  @(AosSchema@field)(implementation = classOf[String], example = "user3")
+  @BeanProperty userDisabled: Option[String] = None,
 
-                        defaultMappingValue: List[DefaultValue] = List(),
```

Comment:
Isn't this a change in the DB model? Thus quite a big change needed a Mongo update script as well.

Reply:
You are correct -- this is probably unwanted. I wanted to get rid of duplicate export fields `defaultMappingValues` and `defaultMappingValue`, that has stayed. Even though using the plural seemed better, it is probably not worth the hassle it would create on the persistence layer (I only imagined the change on the API level, which was bearable) - so I reverted the name to singular as it was.

Thanks for the idea.